src/components/Visualization/useAnalyticsData.js

```diff
@@ -43,8 +43,21 @@ const excludedDimensions = [
     DIMENSION_ID_LAST_UPDATED_BY,
 ]
 
-const findOptionSetItem = (code, metaDataItems) =>
-    Object.values(metaDataItems).find((item) => item.code === code)
+const findOptionSetItem = (optionSetId, code, metaDataItems) => {
+    const optionSetMetaData = metaDataItems?.[optionSetId]
+
+    if (optionSetMetaData) {
+        const optionId = optionSetMetaData.options.find(
+            (option) => option.code === code
+        )?.uid
+
+        if (optionId) {
```

Comment:
Unnecessary check? `metaDataItems[optionId]` will return undefined if not found anyway, so no need to wrap it?

Reply:
I removed the condition.